### ISSUE TITLE
Add playback of captures to webconf

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#40]
Webconf provides a method of managing audio and MIDI recordings (captures). These may be downloaded, renamed and deleted. It can be awkward to manage these files without the context of what each sounds like.

It may prove advantageous to allow replay of each capture file from webconf interface.

I have a patch for this already which I will add as a branch to the repo...

**Comment by riban-bw** [comment]
Branch added to github branch [ghz40_playback](https://github.com/zynthian/zynthian-webconf/tree/ghz40_playback). This uses the html audio tag which does not support MIDI so only working for audio at the moment. It depends on symlinks to the local and USB storage areas so is hard coded to align with zynthian-ui. It seems to work for me and I have managed to test using the new "Repositoires" feature of webconf (cool).

**Comment by wyleu** [comment]
Nice. Works for me.

**Comment by wyleu** [comment]
I have managed to test using the new "Repositoires" feature of webconf (cool).

500's for me with No message in GUI Log.

**Comment by wyleu** [comment]
Seems a new image is required to get repository functionality.

Very nice.

**Comment by jofemodo** [comment]
I would like to have the possibility of playing on zynthian ... 
MIDI play, when available, should play on zynthian too ...

**Comment by jofemodo** [comment]
BTW, i just tested on firefox/linux and it doesn't work. The browser's console tells me (in spanish):

```
URI no válida. Ha fallado la carga del recurso de medios . lib-captures
El recurso de medios http://zynthian.local/captures/zynthian/zynthian-my-data/capture/jack_capture_01.wav no puede decodificarse. lib-captures
No se ha podido decodificar el recurso multimedia http://zynthian.local/captures/zynthian/zynthian-my-data/capture/jack_capture_01.wav, error: Error Code: NS_ERROR_DOM_MEDIA_METADATA_ERR (0x806e0006) lib-captures
```

**Comment by riban-bw** [comment]
Regarding playback on Zynthian, I think that may be a desirable option, i.e. have web playback and separate controls for Zynthian playback. Playback on Zynthian should be relatively easy with a tweak to allow filename to be passed to player.

I will take a look at other browsers later when I finish work.

**Comment by riban-bw** [comment]
I can confirm that Firefox fails to play audio created by Zynthian. It will play other wav files. It seems to be an issue with Firefox not handling RIFF gracefully. Running wav file through sox gives warning:
> wave header missing extended part of fmt chunk
The jack audio recorder seems to be to blame here. I think the right thing to do is to fix (or replace) jack_capture. IE is worse - it does not even recognise valid wav files!

**Comment by wyleu** [comment]
IE is worse - it does not even recognise valid wav files! 
bit ironic when you consider who specified the format.

Presumably it's this ....
Beginning with Windows 2000, a WAVE_FORMAT_EXTENSIBLE header was defined which specifies multiple audio channel data along with speaker positions, eliminates ambiguity regarding sample types and container sizes in the standard WAV format and supports defining custom extensions to the format chunk

**Comment by riban-bw** [comment]
I think we should keep this mod to webconf because in Chrome it really adds a very useful feature. Being able to preview audio right there in the browser whilst managing it (deleting, renaming, etc.) is a real boon. I will look at ways to improve its behaviour for other browsers (currently looking at ways to fix the wav files which are actually broken so would benefit from this fix - I will raise a separate issue about that).

**Comment by wyleu** [comment]
** off topic **
Starts to produces a very nice environment with a musician at the keyboard and an 'engineer' on the webconf. Add record and ogg download and I'll shut up for a bit. :-D

**Comment by riban-bw** [comment]
Firefox support for 24-bit audio was added with commit [14b8a98aff19](https://hg.mozilla.org/mozilla-central/rev/14b8a98aff19) which was targeted at milestone 46.0a1 but I have not found any release notes that say this was implemented. My version of Firefox is 32.0.3 (I know - it is old but not within my gift to update on this machine) so I can't test if 24-bit support has been added.
We may as well show the player for all browsers which will not allow playback of formats the browser does not support (rather than add logic to hide it based on file format and browser compatibility).
@wyleu was playing late at night / early in the morning (I saw your post just before I crashed). There could be an option to create ogg versions of recordings but I suggest you raise a separate ticket for that. It may be quite desirable.
FYI: My effort to add audio recording to libjackpeak proceeds well. I had it recording audio on one channel last night but some changes today have those monaural recordings being distorted. I am getting there though!
